scripts/qa_crowbarsetup.sh

```diff
@@ -3495,17 +3495,26 @@ function oncontroller_upload_defcore
     pushd /var/lib/openstack-tempest-test
     # get the test list
     wget "https://refstack.openstack.org/api/v1/guidelines/2018.02/tests" -O defcore-with-id.txt
+
     # remove the id in [] or tempest will complain on incorrect regex
     sed -e 's/\[[^][]*\]//g' defcore-with-id.txt > defcore.txt
+
     # run only the specified tests
     tempest run --whitelist-file defcore.txt
-    source /root/.openrc
+
+    # get a subunit-2to1 filter directly
     wget https://raw.githubusercontent.com/testing-cabal/subunit/master/filters/subunit-2to1
 
+    # save the latest tempest run results
     local testdriver=testr
     [ -f ".stestr.conf" ] && testdriver=stestr
     $testdriver last --subunit | python subunit-2to1 > tempest.subunit_1.log
-    test -d refstack-client || safely git clone https://github.com/openstack/refstack-client
+
+    # clone a refstack client
+    test -d refstack-client || safely git clone https://github.com/kbaikov/refstack-client
+
+    # upload to refstack server
+    source /root/.openrc
```

Comment:
Why source this here? Is it generated by `refstack-client`?

Reply:
This is later used as a parameter --keystone-endpoint $OS_AUTH_URL. This is our standard openrc file.

Reply:
But you moved the line, so I was wondering why :smile:

Reply:
I moved it right before the place it was needed. As i added one line comments and separated lines to logical blocks. So no particular reason for that line to be exactly there.

Reply:
Ah, makes sense. Thanks!